lib/api/multiObjectDelete.js

```diff
@@ -166,6 +169,60 @@ function _parseXml(xmlToParse, next) {
     });
 }
 
+/**
+ * processObjectVersion - process object version to be deleted
+ * @param {object} entry - entry from data model
+ * @param {string} bucketName - bucket name
+ * @param {function} next - callback to call with error or decoded version
+ * @return {undefined}
+ **/
+function processObjectVersion(entry, bucketName) {
+    let decodedVersionId;
+    if (entry.versionId) {
+        decodedVersionId = entry.versionId === 'null' ?
+            'null' : versionIdUtils.decode(entry.versionId);
+    }
+    if (decodedVersionId instanceof Error) {
+        monitoring.promMetrics('DELETE', bucketName, 404,
+            'multiObjectDelete');
+        return [errors.NoSuchVersion];
+    }
+    return [null, decodedVersionId];
+}
+
+/**
+ * Initialization function for the MultiObjectDelete API that will, based on the
+ * current metadata backend,a ssess if metadata READ batching is supported. If
+ * yes, the initialization step will call the metadataGetObjects function from
+ * the MetadataWrapper.
+ * @param {string} bucketName - bucket name
+ * @param {string []} inPlay - list of object keys still in play
+ * @param {object} log - logger object
+ * @return {undefined}
+ */
+function shouldBatchGetObjectsFromMetadata(bucketName, inPlay, log) {
+    const backendSupportsBatching = constants.supportsBatchingMethods.includes(config.backends.metadata);
+    // If the backend supports batching, we want to optimize the API latency by
+    // first getting all the objects metadata, stored in memory, for later use
+    // in the API. This approach does not change the API architecture, but
+    // transplants an additional piece of code that can greatly improve the API
+    // latency when the database supports batching.
+    if (backendSupportsBatching) {
+        const objectKeys = Object.keys(inPlay).map(entry => {
```

Comment:
```suggestion
        const objectKeys = Object.values(inPlay).map(entry => {
```

And then use `entry` directly instead of `inPlay[entry]`